### ISSUE TITLE
Feature request: little/big endian output

### ISSUE THREAD

**Issue by mikrosk** [libxmp/libxmp#679]
Would it be possible to add an option (similar to 8/16-bit, signed/unsigned etc) for explicitly outputting big/little endian samples? Right now it's always platforms native format (which makes sense) but I have a few use cases where this would be super-useful.

**Comment by sezero** [comment]
I don't think that we want the extra complexity involved with that, but let's hear what others have to say: @cmatsuoka, @AliceLR?

**Comment by sagamusix** [comment]
I don't think this makes any sense on the API level. If you need specific endianness, you can always post-process libxmp's output (which is what libxmp would have to do itself anyway so it's not like this would allow for optimizations that you couldn't do in your own application).

**Comment by mikrosk** [comment]
> you can always post-process libxmp's output (which is what libxmp would have to do itself anyway

Of course, if this is the way it would have been implemented then it makes no sense. What I had in mind is a macro or inline function which would according to a flag either write the sample as is (i.e. *buffer++ = generated_sample) or swap the byte order on the fly. That way it would be much faster than doing post-processing afterwards.

**Comment by sezero** [comment]
Closing as WONTFIX
